pkg/collector/corechecks/snmp/internal/devicecheck/devicecheck.go

```diff
@@ -249,17 +249,27 @@ func (d *DeviceCheck) detectAvailableMetrics() ([]checkconfig.MetricsConfig, []c
 	var metricConfigs []checkconfig.MetricsConfig
 	var metricTagConfigs []checkconfig.MetricTagConfig
 
+	// If a metric name has already been encountered, we won't try to add it again.
+	alreadySeenMetrics := make(map[string]bool)
```

Comment:
<!-- Nitpicks are small, trivial, but necessary changes. Distinguishing nitpick comments significantly helps direct the reader's attention to comments requiring more involvement. -->
**🥜 nitpick**
```suggestion
	alreadySeenMetrics := make(map[string]struct{})
```

Reply:
Indeed, it's an alternative and it will use less memory.

but I have a preference for `bool` since it makes usage a bit easier.

Using bool you can:
```
if !alreadySeenMetrics[metricName] && root.LeafExist(metricConfig.Symbol.OID) {
```

Using struct you need:
```
_, ok := alreadySeenMetrics[metricName]
if !ok && root.LeafExist(metricConfig.Symbol.OID) {
```